modules/exploits/windows/http/northstar_c2_xss_to_agent_rce.rb

```diff
@@ -0,0 +1,237 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer::HTML
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'NorthStar C2 XSS to Agent RCE',
+        'Description' => %q{
+          NorthStar C2, prior to commit 7674a44 on March 11 2024, contains a vulnerability where the logs page is
+          vulnerable to a stored xss.
+          An unauthenticated user can simulate an agent registration to cause the XSS and take over a users session.
+          With this access, it is then possible to run a new payload on all of the NorthStar C2 compromised hosts
+          (agents), and kill the original agent.
+          Successfully tested against NorthStar C2 commit e7fdce148b6a81516e8aa5e5e037acd082611f73 running on
+          Ubuntu 22.04. The agent was running on Windows 10 19045.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'chebuya' # original PoC, analysis
+        ],
+        'DefaultOptions' => {
+          'WfsDelay' => 3_600, # 1hr
```

Comment:
Could this be a passive module instead that executes and serves the payload in the background?